### PR TITLE
Update frontend code documentation with v2.0 changes

### PR DESCRIPTION
**Pull request by maryamkhidir**
### Feature or Bugfix

- Feature

### Detail
- Update frontend code documentation with v2.0 changes

### Relates
- https://github.com/awslabs/aws-dataall/pull/654

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

 N/A
- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pages/code.md [hunk 1]
```diff
@@ -632,90 +659,204 @@ class MLStudioApiModuleInterface(ModuleInterface):
...
+├── src/ : contains the major components of the frontend code, to be discussed in detail in the src/ section below
+└── jsconfig.json : used to reference the root folder and map aliases/modules to their respective paths relative to the root folder.
+```
+
+### src/ <a name="src"></a>
+This section contains the major components of the frontend code. Here is a short description of all the components of the src folder, we will deep dive subsequently on the contents of each module.
+
+```
+src/
+├── authentication/ : contains files, contexts, hooks related to user and guest authentication
```
Comment (dlpzx) [inline pages/code.md:684]: I don't think we have any "guest" authentication

File: pages/code.md [hunk 2]
```diff
@@ -632,90 +659,204 @@ class MLStudioApiModuleInterface(ModuleInterface):
...
+├── hooks/
+├── views/
+└── index.js
+```
+
+We used React Context API to handle the authentication state mangagement. `CognitoAuthContext.js` handles the AWS deployment and `LocalAuthContext.js` for local deployment.
+Then `AWS Amplify` is used to connect the app to `AWS Cognito` for production auth and a default anonymous user  is set for local environment.
+
+The `useAuth` hook is used to decide the auth context to use depending on the deployment environment, and `AuthGuard` is a wrapper to verify authentication before routing users to the requested pages in the application.
+
```
Comment (dlpzx) [inline pages/code.md:710]: I think lines 711 to 714 should belong to other parts of the documentation (it is more like user guide, getting started...) here I would focus on code only

**Review by dlpzx — APPROVED** [review]
Looks very good, thank you! I added 2 very minor comments. Also, should we mention how we read the config.json and how enabling/disabling modules works?

File: pages/code.md (hunk 2 above)
Comment (maryamkhidir) [reply pages/code.md:710]: Okay this will be refactored out

Comment (maryamkhidir) [reply pages/code.md:710]: updates added here https://github.com/awslabs/aws-dataall/pull/722/commits/76b8b7b3178a65553b88b72164562121b76bfde5

**Pull request merged**